src/renderers/abstract.renderer.ts

```diff
@@ -772,17 +772,17 @@ export abstract class Renderer {
     })
 
     /* Remove empty paragraphs */
-    // TODO: Refactor this option to work with page/html and page/mobile-html output. See issues/1866
     if (!dump.opts.keepEmptyParagraphs) {
-      // Mobile view === details
-      // Desktop view === section
-      const sections: DominoElement[] = Array.from(parsoidDoc.querySelectorAll('details, section'))
+      const sections: DominoElement[] = Array.from(parsoidDoc.querySelectorAll('section'))
```

Comment:
This won't do anything with ActionParse for wikis that don't use Parsoid.

Reply:
Yes, I let @kelson42 confirm but from my understanding this is "accepted" in the sense that this feature is anyway mostly only for wikimedia wikis which are using Parsoid. And we have no real alternative for other wikis (besides very complex and fragile logic) since all HTML is "at the same level" without proper sections.

Reply:
@kelson42 your point makes no sense ; for wikis which don't use Parsoid, the code will not trigger so we will always keep empty paragraphs. Passing the `--keepEmptyParagraphs` will not change anything. And it is not really possible to detect that we've kept empty paragraphs since we don't know these are empty paragraphs otherwise we would have deleted them.

Reply:
I would not bother users about having to set a flag to get the same behavior than if the flag is not set. Of course, users won't know. But they don't know about this feature anyway in the first place.

Reply:
We have to  bother users if the software does not deliver like expected. We don't do things behind the scene without clearly inform users... and just a warning is IMO not enough. Anyway, not a deal breaker for me. At least put a clear warning.

Reply:
> At least put a clear warning

I have no clue how to do this, especially reliably. I've moved this topic to https://github.com/openzim/mwoffliner/issues/2431 so that we can at least close this PR. 

@kelson42 please review